packages/analyticsdx-template-lint/src/linter.ts

```diff
@@ -66,17 +66,26 @@ function findAllItemsForLayoutDefinition(layoutJson: JsonNode): JsonNode[] {
  */
 function findAllVariableNamesForLayoutDefinition(layoutJson: JsonNode): Array<{ name: string; nameNode: JsonNode }> {
   return findAllItemsForLayoutDefinition(layoutJson).reduce((items, item) => {
-    const type = findJsonPrimitiveAttributeValue(item, 'type')[0];
-    if (type === 'Variable') {
-      const [name, nameNode] = findJsonPrimitiveAttributeValue(item, 'name');
-      if (typeof name === 'string' && nameNode) {
-        items.push({ name, nameNode });
-      }
-    }
-    return items;
+    const variableItems = findAllVariableItemsForLayoutDefinition(item);
+    return [...items, ...variableItems];
```

Comment:
Nit: could do this to avoid creating a new outer array each time:
```suggestion
    items.push(...variableItems);
    return items;
```